flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java

```diff
@@ -216,6 +227,49 @@ public WebMonitorEndpoint(
 
         this.leaderElectionService = Preconditions.checkNotNull(leaderElectionService);
         this.fatalErrorHandler = Preconditions.checkNotNull(fatalErrorHandler);
+
+        this.threadInfoOperatorTracker = initializeThreadInfoTracker(executor);
+    }
+
+    private ThreadInfoOperatorTracker<OperatorThreadInfoStats> initializeThreadInfoTracker(
+            ScheduledExecutorService executor) {
+        final Duration akkaTimeout;
+        try {
+            akkaTimeout = AkkaUtils.getTimeout(clusterConfiguration);
+        } catch (NumberFormatException e) {
+            throw new IllegalConfigurationException(AkkaUtils.formatDurationParsingErrorMessage());
+        }
+
+        final int flameGraphCleanUpInterval =
+                clusterConfiguration.getInteger(WebOptions.FLAMEGRAPH_CLEANUP_INTERVAL);
+        final ThreadInfoRequestCoordinator threadInfoRequestCoordinator =
+                new ThreadInfoRequestCoordinator(executor, akkaTimeout.toMillis());
+        final ThreadInfoOperatorTracker<OperatorThreadInfoStats> threadInfoOperatorTracker =
+                ThreadInfoOperatorTracker.newBuilder(
+                                resourceManagerRetriever, Function.identity(), executor)
+                        .setCoordinator(threadInfoRequestCoordinator)
+                        .setCleanUpInterval(flameGraphCleanUpInterval)
+                        .setNumSamples(
+                                clusterConfiguration.getInteger(WebOptions.FLAMEGRAPH_NUM_SAMPLES))
+                        .setStatsRefreshInterval(
+                                clusterConfiguration.getInteger(
+                                        WebOptions.FLAMEGRAPH_REFRESH_INTERVAL))
+                        .setDelayBetweenSamples(
+                                Time.milliseconds(
+                                        clusterConfiguration.getInteger(
+                                                WebOptions.FLAMEGRAPH_DELAY)))
+                        .setMaxThreadInfoDepth(
+                                clusterConfiguration.getInteger(
+                                        WebOptions.FLAMEGRAPH_STACK_TRACE_DEPTH))
+                        .build();
+
+        executor.scheduleWithFixedDelay(
```

Comment:
shouldn't the tracker care of care of initiating this cleanup?

Reply:
I understand this as:

1. pass in the `executor` as it is done now
2. schedule the cleanup in the constructor of the tracker.

Right?

Reply:
yes

Reply:
Done.